### PR TITLE
[frontend] fix Distribution of Opinions if no opinion exist (#5306)

### LINKED ISSUE #5306: "An unknown error occurred..." in distribution of opinions of a report
## Description

When you click on "Distribution of opinions" of a report (see pictures below), the "An unknown error occurred..." panel appears.

## Environment

1. OS (where OpenCTI server runs): macOS Sonoma 14.2.1
2. OpenCTI version: 5.12.13
3. OpenCTI client: frontend with Firefox 121
4. Other environment details:

## Reproducible Steps

Steps to create the smallest reproducible scenario:
1. Select a report in the list of reports
2. Click on "Distribution of opinions"

## Expected Output

The distribution of opinions concerning the selected report.

## Actual Output

Error
An unknown error occurred. Please contact your administrator or the OpenCTI maintainers.

## Screenshots (optional)

The selected report

image

The result after clicking on "Distribution of opinions"

image

image

### PR DESCRIPTION
**Pull request by Archidoit**
issue: https://github.com/OpenCTI-Platform/opencti/issues/5306

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/analyses/opinions/StixCoreObjectOpinionsRadarDialog.tsx [hunk 1]
```diff
@@ -81,7 +83,21 @@ StixCoreObjectOpinionsRadarDialogProps
...
+  const handleOpen = () => {
+    if (opinionOptions.length > 0) {
+      setOpen(true);
+    } else {
+      MESSAGING$.notifyError(
+        <span>
+          {t('There are no opinions. You can create some ')}{' '}
+          <Link to="/dashboard/settings/vocabularies/fields/opinion_ov">
+            {t('in this page')}
+          </Link>
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/private/components/analyses/opinions/StixCoreObjectOpinionsRadarDialog.tsx]: It seems we miss translations here

Comment (lndrtrbn) [reply opencti-platform/opencti-front/src/private/components/analyses/opinions/StixCoreObjectOpinionsRadarDialog.tsx]: And are we sure we want to add a link to an admin page when majority of people won't have access to it?

**Review by lndrtrbn — COMMENTED** [review]
Tested locally, no crash when no vocab in ES

File: opencti-platform/opencti-front/src/private/components/analyses/opinions/StixCoreObjectOpinionsRadarDialog.tsx (hunk 1 above)
Comment (Archidoit) [reply opencti-platform/opencti-front/src/private/components/analyses/opinions/StixCoreObjectOpinionsRadarDialog.tsx]: mmm yes you're right, I will rather add a warning message with no redirection link.

Comment (Archidoit) [reply opencti-platform/opencti-front/src/private/components/analyses/opinions/StixCoreObjectOpinionsRadarDialog.tsx]: fixed

File: opencti-platform/opencti-front/src/utils/Localization.js [hunk 2]
```diff
@@ -1599,6 +1599,7 @@ const i18n = {
       'Distribution of entities': 'Distribución de entidades',
       'Entities distribution': 'Distribución de entidades',
       'Distribution of opinions': 'Distribución de opiniones',
+      'There are no opinions. You should create some before.': 'There are no opinions. You should create some before.',
```
Comment (lndrtrbn) [inline opencti-platform/opencti-front/src/utils/Localization.js]:
```suggestion
      'There are no opinions. You should create some before.': 'No hay opiniones. Deberías crear algunos antes.',
```

**Review by lndrtrbn — CHANGES_REQUESTED** [review]

**Comment by Archidoit** [comment]
@lndrtrbn  the comments have been adressed :)

**Review by lndrtrbn — APPROVED** [review]

**Pull request merged**